packages/discord-ticket-api/src/queue.ts

```diff
@@ -1,39 +1,96 @@
+    // Node default is 0 (disabled) in many environments.
+    keepAlive: 30_000,
+    connectTimeout: 10_000,
     maxRetriesPerRequest: null,
     enableReadyCheck: true,
+    // Back off reconnects a bit; ioredis will reconnect by default, but without this it can spam.
+    retryStrategy: (times) => Math.min(times * 50, 2_000),
     ...tls,
   };
 }
 
-export function createQueue(env: AppEnv, logger: FastifyBaseLogger): Queue<TicketJobData, unknown, string> {
-  const connection = new Redis(createRedisConnection(env));
-  const opts: QueueOptions = {
-    connection,
-  };
+export type TicketQueue = {
+  queue: Queue<TicketJobData, unknown, string>;
+  events: QueueEvents;
+  close: () => Promise<void>;
+};
+
+function attachRedisLogging(connection: IORedisClient, logger: FastifyBaseLogger, name: string, env: AppEnv) {
+  connection.on("error", (err) => {
+    logger.warn(
+      { err, redis: { host: env.REDIS_HOST, port: env.REDIS_PORT }, connection: name },
+      "Redis connection error"
+    );
+  });
+  connection.on("close", () => {
+    logger.warn({ redis: { host: env.REDIS_HOST, port: env.REDIS_PORT }, connection: name }, "Redis connection closed");
+  });
+  connection.on("reconnecting", () => {
+    logger.info(
+      { redis: { host: env.REDIS_HOST, port: env.REDIS_PORT }, connection: name },
+      "Redis connection reconnecting"
+    );
+  });
+  connection.on("end", () => {
+    logger.warn({ redis: { host: env.REDIS_HOST, port: env.REDIS_PORT }, connection: name }, "Redis connection ended");
+  });
+}
+
+export function createQueue(env: AppEnv, logger: FastifyBaseLogger): TicketQueue {
+  const connectionOpts = createRedisConnection(env);
+  const queueConnection = new Redis(connectionOpts);
+  const eventsConnection = new Redis(connectionOpts);
+
+  attachRedisLogging(queueConnection, logger, "queue", env);
+  attachRedisLogging(eventsConnection, logger, "events", env);
+
+  const opts: QueueOptions = { connection: queueConnection };
   const queue = new Queue<TicketJobData, unknown, string>(env.QUEUE_NAME, opts);
-  const events = new QueueEvents(env.QUEUE_NAME, { connection });
+
+  // BullMQ recommends using a dedicated connection for QueueEvents.
+  const events = new QueueEvents(env.QUEUE_NAME, { connection: eventsConnection });
   events.on("failed", ({ jobId, failedReason }) => {
     logger.error({ jobId, failedReason }, "Queue job failed");
   });
   events.on("completed", ({ jobId }) => {
     logger.info({ jobId }, "Queue job completed");
   });
-  return queue;
+  events.on("error", (err: unknown) => {
+    logger.warn({ err, queue: env.QUEUE_NAME }, "QueueEvents error");
+  });
+
+  // Queue can also surface redis errors; log them to avoid uncaught/no-context stack traces.
+  // (Some error conditions originate from underlying Redis sockets, e.g. ECONNRESET.)
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (queue as any).on?.("error", (err: unknown) => {
```

Comment:
is casting to `unknown` needed? My IDL infers it as `Error`

Also why casting `queue` as `any`?